--- 1/Config.in
+++ 2/Config.in
@@ -111,6 +111,7 @@
 config PAM
 	bool "Support PAM (Pluggable Authentication Modules)"
 	default n
+	depends on BROKEN_ON_HURD
 	help
 	Use PAM in some applets (currently login and httpd) instead
 	of direct access to password database.
@@ -118,6 +119,7 @@
 config FEATURE_DEVPTS
 	bool "Use the devpts filesystem for Unix98 PTYs"
 	default y
+	depends on BROKEN_ON_HURD
 	help
 	Enable if you want to use Unix98 PTY support. If enabled,
 	busybox will use /dev/ptmx for the master side of the pseudoterminal
@@ -313,6 +315,7 @@
 	bool "Support NSA Security Enhanced Linux"
 	default n
 	select PLATFORM_LINUX
+	depends on BROKEN_ON_HURD
 	help
 	Enable support for SELinux in applets ls, ps, and id. Also provide
 	the option of compiling in SELinux applets.
@@ -369,6 +372,10 @@
 	#This is automatically selected if any applet or feature requires
 	#Linux-specific interfaces. You do not need to select it manually.
 
+config BROKEN_ON_HURD
+	bool
+	default n
+
 comment 'Build Options'
 
 config STATIC
--- 1/console-tools/deallocvt.c
+++ 2/console-tools/deallocvt.c
@@ -11,6 +11,7 @@
 //config:	bool "deallocvt (1.9 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	This program deallocates unused virtual consoles.
 
--- 1/console-tools/fgconsole.c
+++ 2/console-tools/fgconsole.c
@@ -10,6 +10,7 @@
 //config:	bool "fgconsole (1.5 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	This program prints active (foreground) console number.
 
--- 1/console-tools/kbd_mode.c
+++ 2/console-tools/kbd_mode.c
@@ -12,6 +12,7 @@
 //config:	bool "kbd_mode (4.1 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	This program reports and sets keyboard mode.
 
--- 1/console-tools/loadfont.c
+++ 2/console-tools/loadfont.c
@@ -13,6 +13,7 @@
 //config:	bool "loadfont (5.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	This program loads a console font from standard input.
 //config:
@@ -20,6 +21,7 @@
 //config:	bool "setfont (24 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:       depends on BROKEN_ON_HURD
 //config:	help
 //config:	Allows to load console screen map. Useful for i18n.
 //config:
--- 1/console-tools/openvt.c
+++ 2/console-tools/openvt.c
@@ -11,6 +11,7 @@
 //config:	bool "openvt (7.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	This program is used to start a command on an unused
 //config:	virtual terminal.
--- 1/console-tools/setlogcons.c
+++ 2/console-tools/setlogcons.c
@@ -12,6 +12,7 @@
 //config:	bool "setlogcons (1.8 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	This program redirects the output console of kernel messages.
 
--- 1/console-tools/showkey.c
+++ 2/console-tools/showkey.c
@@ -10,6 +10,7 @@
 //config:	bool "showkey (4.7 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Shows keys pressed.
 
--- 1/coreutils/nproc.c
+++ 2/coreutils/nproc.c
@@ -6,6 +6,7 @@
 //config:config NPROC
 //config:	bool "nproc (3.7 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Print number of CPUs
 
--- 1/e2fsprogs/tune2fs.c
+++ 2/e2fsprogs/tune2fs.c
@@ -9,6 +9,7 @@
 //config:config TUNE2FS
 //config:	bool "tune2fs (4.4 kb)"
 //config:	default n  # off: it is too limited compared to upstream version
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	tune2fs allows the system administrator to adjust various tunable
 //config:	filesystem parameters on Linux ext2/ext3 filesystems.
--- 1/init/bootchartd.c
+++ 2/init/bootchartd.c
@@ -5,6 +5,7 @@
 //config:config BOOTCHARTD
 //config:	bool "bootchartd (10 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	bootchartd is commonly used to profile the boot process
 //config:	for the purpose of speeding it up. In this case, it is started
--- 1/klibc-utils/run-init.c
+++ 2/klibc-utils/run-init.c
@@ -9,6 +9,7 @@
 //config:	bool "run-init (7.7 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The run-init utility is used from initramfs to select a new
 //config:	root device. Under initramfs, you have to use this instead of
--- 1/libbb/common_bufsiz.c
+++ 2/libbb/common_bufsiz.c
@@ -9,6 +9,7 @@
 //config:config FEATURE_USE_BSS_TAIL
 //config:	bool "Use the end of BSS page"
 //config:	default n
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Attempt to reclaim a small unused part of BSS.
 //config:
--- 1/mailutils/popmaildir.c
+++ 2/mailutils/popmaildir.c
@@ -12,6 +12,7 @@
 //config:config POPMAILDIR
 //config:	bool "popmaildir (10 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Simple yet powerful POP3 mail popper. Delivers content
 //config:	of remote mailboxes to local Maildir.
--- 1/mailutils/sendmail.c
+++ 2/mailutils/sendmail.c
@@ -9,6 +9,7 @@
 //config:config SENDMAIL
 //config:	bool "sendmail (14 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Barebones sendmail.
 
--- 1/miscutils/adjtimex.c
+++ 2/miscutils/adjtimex.c
@@ -14,6 +14,7 @@
 //config:	bool "adjtimex (4.7 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Adjtimex reads and optionally sets adjustment parameters for
 //config:	the Linux clock adjustment algorithm.
--- 1/miscutils/beep.c
+++ 2/miscutils/beep.c
@@ -10,6 +10,7 @@
 //config:	bool "beep (2.4 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The beep applets beeps in a given freq/Hz.
 //config:
--- 1/miscutils/conspy.c
+++ 2/miscutils/conspy.c
@@ -13,6 +13,7 @@
 //config:	bool "conspy (10 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	A text-mode VNC like program for Linux virtual terminals.
 //config:	example:  conspy NUM      shared access to console num
--- 1/miscutils/fbsplash.c
+++ 2/miscutils/fbsplash.c
@@ -24,6 +24,7 @@
 //config:	bool "fbsplash (26 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Shows splash image and progress bar on framebuffer device.
 //config:	Can be used during boot phase of an embedded device.
--- 1/miscutils/flash_eraseall.c
+++ 2/miscutils/flash_eraseall.c
@@ -13,6 +13,7 @@
 //config:config FLASH_ERASEALL
 //config:	bool "flash_eraseall (5.9 kb)"
 //config:	default n  # doesn't build on Ubuntu 8.04
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The flash_eraseall binary from mtd-utils as of git head c4c6a59eb.
 //config:	This utility is used to erase the whole MTD device.
--- 1/miscutils/flash_lock_unlock.c
+++ 2/miscutils/flash_lock_unlock.c
@@ -7,6 +7,7 @@
 //config:config FLASH_LOCK
 //config:	bool "flash_lock (2.1 kb)"
 //config:	default n  # doesn't build on Ubuntu 8.04
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The flash_lock binary from mtd-utils as of git head 5ec0c10d0. This
 //config:	utility locks part or all of the flash device.
@@ -14,6 +15,7 @@
 //config:config FLASH_UNLOCK
 //config:	bool "flash_unlock (1.3 kb)"
 //config:	default n  # doesn't build on Ubuntu 8.04
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The flash_unlock binary from mtd-utils as of git head 5ec0c10d0. This
 //config:	utility unlocks part or all of the flash device.
--- 1/miscutils/flashcp.c
+++ 2/miscutils/flashcp.c
@@ -9,6 +9,7 @@
 //config:config FLASHCP
 //config:	bool "flashcp (5.3 kb)"
 //config:	default n  # doesn't build on Ubuntu 8.04
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The flashcp binary, inspired by mtd-utils as of git head 5eceb74f7.
 //config:	This utility is used to copy images into a MTD device.
--- 1/miscutils/hdparm.c
+++ 2/miscutils/hdparm.c
@@ -15,6 +15,7 @@
 //config:	bool "hdparm (25 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Get/Set hard drive parameters. Primarily intended for ATA
 //config:	drives.
--- 1/miscutils/i2c_tools.c
+++ 2/miscutils/i2c_tools.c
@@ -12,6 +12,7 @@
 //config:	bool "i2cget (5.5 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Read from I2C/SMBus chip registers.
 //config:
@@ -19,6 +20,7 @@
 //config:	bool "i2cset (6.7 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Set I2C registers.
 //config:
@@ -26,6 +28,7 @@
 //config:	bool "i2cdump (7.1 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Examine I2C registers.
 //config:
@@ -33,6 +36,7 @@
 //config:	bool "i2cdetect (7.1 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Detect I2C chips.
 //config:
@@ -40,6 +44,7 @@
 //config:	bool "i2ctransfer (4.0 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Send user-defined I2C messages in one transfer.
 //config:
--- 1/miscutils/inotifyd.c
+++ 2/miscutils/inotifyd.c
@@ -29,6 +29,7 @@
 //config:config INOTIFYD
 //config:	bool "inotifyd (3.6 kb)"
 //config:	default n  # doesn't build on Knoppix 5
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Simple inotify daemon. Reports filesystem changes. Requires
 //config:	kernel >= 2.6.13
--- 1/miscutils/nandwrite.c
+++ 2/miscutils/nandwrite.c
@@ -11,6 +11,7 @@
 //config:	bool "nandwrite (4.8 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Write to the specified MTD device, with bad blocks awareness
 //config:
@@ -18,6 +19,7 @@
 //config:	bool "nanddump (5.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Dump the content of raw NAND chip
 
--- 1/miscutils/partprobe.c
+++ 2/miscutils/partprobe.c
@@ -8,6 +8,7 @@
 //config:	bool "partprobe (3.5 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Ask kernel to rescan partition table.
 
--- 1/miscutils/raidautorun.c
+++ 2/miscutils/raidautorun.c
@@ -10,6 +10,7 @@
 //config:	bool "raidautorun (1.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	raidautorun tells the kernel md driver to
 //config:	search and start RAID arrays.
--- 1/miscutils/readahead.c
+++ 2/miscutils/readahead.c
@@ -14,6 +14,7 @@
 //config:	default y
 //config:	depends on LFS
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Preload the files listed on the command line into RAM cache so that
 //config:	subsequent reads on these files will not block on disk I/O.
--- 1/miscutils/rfkill.c
+++ 2/miscutils/rfkill.c
@@ -10,6 +10,7 @@
 //config:	bool "rfkill (4.4 kb)"
 //config:	default n # doesn't build on Ubuntu 9.04
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Enable/disable wireless devices.
 //config:
--- 1/miscutils/setserial.c
+++ 2/miscutils/setserial.c
@@ -11,6 +11,7 @@
 //config:	bool "setserial (6.9 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Retrieve or set Linux serial port.
 
--- 1/miscutils/ubi_tools.c
+++ 2/miscutils/ubi_tools.c
@@ -7,6 +7,7 @@
 //config:	bool "ubiattach (4.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Attach MTD device to an UBI device.
 //config:
@@ -14,6 +15,7 @@
 //config:	bool "ubidetach (4.1 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Detach MTD device from an UBI device.
 //config:
@@ -21,6 +23,7 @@
 //config:	bool "ubimkvol (5.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Create a UBI volume.
 //config:
@@ -28,6 +31,7 @@
 //config:	bool "ubirmvol (4.9 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Delete a UBI volume.
 //config:
@@ -35,6 +39,7 @@
 //config:	bool "ubirsvol (4.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Resize a UBI volume.
 //config:
@@ -42,6 +47,7 @@
 //config:	bool "ubiupdatevol (5.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Update a UBI volume.
 
--- 1/miscutils/ubirename.c
+++ 2/miscutils/ubirename.c
@@ -10,6 +10,7 @@
 //config:	bool "ubirename (2.4 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Utility to rename UBI volumes
 
--- 1/miscutils/watchdog.c
+++ 2/miscutils/watchdog.c
@@ -12,6 +12,7 @@
 //config:	bool "watchdog (5.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The watchdog utility is used with hardware or software watchdog
 //config:	device drivers. It opens the specified watchdog device special file
--- 1/modutils/depmod.c
+++ 2/modutils/depmod.c
@@ -11,6 +11,7 @@
 //config:	bool "depmod (27 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	depmod generates modules.dep (and potentially modules.alias
 //config:	and modules.symbols) that contain dependency information
--- 1/modutils/insmod.c
+++ 2/modutils/insmod.c
@@ -10,6 +10,7 @@
 //config:	bool "insmod (22 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	insmod is used to load specified modules in the running kernel.
 
--- 1/modutils/lsmod.c
+++ 2/modutils/lsmod.c
@@ -11,6 +11,7 @@
 //config:	bool "lsmod (1.9 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	lsmod is used to display a list of loaded modules.
 //config:
--- 1/modutils/modinfo.c
+++ 2/modutils/modinfo.c
@@ -9,6 +9,7 @@
 //config:	bool "modinfo (24 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Show information about a Linux Kernel module
 
--- 1/modutils/modprobe.c
+++ 2/modutils/modprobe.c
@@ -11,6 +11,7 @@
 //config:	bool "modprobe (28 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Handle the loading of modules, and their dependencies on a high
 //config:	level.
--- 1/modutils/rmmod.c
+++ 2/modutils/rmmod.c
@@ -11,6 +11,7 @@
 //config:	bool "rmmod (3.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	rmmod is used to unload specified modules from the kernel.
 
--- 1/networking/arp.c
+++ 2/networking/arp.c
@@ -16,6 +16,7 @@
 //config:	bool "arp (10 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Manipulate the system ARP cache.
 
--- 1/networking/arping.c
+++ 2/networking/arping.c
@@ -8,6 +8,7 @@
 //config:config ARPING
 //config:	bool "arping (9 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	select PLATFORM_LINUX
 //config:	help
 //config:	Ping hosts by ARP packets.
--- 1/networking/brctl.c
+++ 2/networking/brctl.c
@@ -13,6 +13,7 @@
 //config:	bool "brctl (4.7 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Manage ethernet bridges.
 //config:	Supports addbr/delbr and addif/delif.
--- 1/networking/ether-wake.c
+++ 2/networking/ether-wake.c
@@ -67,6 +67,7 @@
 //config:	bool "ether-wake (4.9 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Send a magic packet to wake up sleeping machines.
 
--- 1/networking/ifconfig.c
+++ 2/networking/ifconfig.c
@@ -28,6 +28,7 @@
 //config:	bool "ifconfig (12 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Ifconfig is used to configure the kernel-resident network interfaces.
 //config:
--- 1/networking/ifenslave.c
+++ 2/networking/ifenslave.c
@@ -101,6 +101,7 @@
 //config:	bool "ifenslave (13 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Userspace application to bind several interfaces
 //config:	to a logical interface (use with kernel bonding driver).
--- 1/networking/ifplugd.c
+++ 2/networking/ifplugd.c
@@ -10,6 +10,7 @@
 //config:	bool "ifplugd (10 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Network interface plug detection daemon.
 
--- 1/networking/ip.c
+++ 2/networking/ip.c
@@ -12,6 +12,7 @@
 //config:	bool "ip (35 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The "ip" applet is a TCP/IP interface configuration and routing
 //config:	utility.
@@ -24,6 +25,7 @@
 //config:	default y
 //config:	select FEATURE_IP_ADDRESS
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Short form of "ip addr"
 //config:
@@ -32,6 +34,7 @@
 //config:	default y
 //config:	select FEATURE_IP_LINK
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Short form of "ip link"
 //config:
@@ -40,6 +43,7 @@
 //config:	default y
 //config:	select FEATURE_IP_ROUTE
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Short form of "ip route"
 //config:
@@ -48,6 +52,7 @@
 //config:	default y
 //config:	select FEATURE_IP_TUNNEL
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Short form of "ip tunnel"
 //config:
@@ -56,6 +61,7 @@
 //config:	default y
 //config:	select FEATURE_IP_RULE
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Short form of "ip rule"
 //config:
@@ -64,6 +70,7 @@
 //config:	default y
 //config:	select FEATURE_IP_NEIGH
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Short form of "ip neigh"
 //config:
--- 1/networking/nameif.c
+++ 2/networking/nameif.c
@@ -14,6 +14,7 @@
 //config:	default y
 //config:	select PLATFORM_LINUX
 //config:	select FEATURE_SYSLOG
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	nameif is used to rename network interface by its MAC address.
 //config:	Renamed interfaces MUST be in the down state.
--- 1/networking/nbd-client.c
+++ 2/networking/nbd-client.c
@@ -6,6 +6,7 @@
 //config:config NBDCLIENT
 //config:	bool "nbd-client (6 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Network block device client
 
--- 1/networking/ntpd.c
+++ 2/networking/ntpd.c
@@ -44,6 +44,7 @@
 //config:	bool "ntpd (22 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The NTP client/server daemon.
 //config:
--- 1/networking/route.c
+++ 2/networking/route.c
@@ -28,6 +28,7 @@
 //config:	bool "route (8.7 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Route displays or manipulates the kernel's IP routing tables.
 
--- 1/networking/slattach.c
+++ 2/networking/slattach.c
@@ -16,6 +16,7 @@
 //config:	bool "slattach (6.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	slattach configures serial line as SLIP network interface.
 
--- 1/networking/tc.c
+++ 2/networking/tc.c
@@ -9,6 +9,7 @@
 //config:config TC
 //config:	bool "tc (8.3 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Show / manipulate traffic control settings
 //config:
--- 1/networking/tunctl.c
+++ 2/networking/tunctl.c
@@ -13,6 +13,7 @@
 //config:	bool "tunctl (6.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	tunctl creates or deletes tun devices.
 //config:
--- 1/networking/udhcp/Config.src
+++ 2/networking/udhcp/Config.src
@@ -7,6 +7,7 @@
 	bool "udhcpd (21 kb)"
 	default y
 	select PLATFORM_LINUX
+	depends on BROKEN_ON_HURD
 	help
 	udhcpd is a DHCP server geared primarily toward embedded systems,
 	while striving to be fully functional and RFC compliant.
@@ -54,6 +55,7 @@
 config DHCPRELAY
 	bool "dhcprelay (5.2 kb)"
 	default y
+	depends on BROKEN_ON_HURD
 	help
 	dhcprelay listens for DHCP requests on one or more interfaces
 	and forwards these requests to a different interface or DHCP
@@ -63,6 +65,7 @@
 	bool "udhcpc (24 kb)"
 	default y
 	select PLATFORM_LINUX
+	depends on BROKEN_ON_HURD
 	help
 	udhcpc is a DHCP client geared primarily toward embedded systems,
 	while striving to be fully functional and RFC compliant.
--- 1/networking/udhcp/d6_dhcpc.c
+++ 2/networking/udhcp/d6_dhcpc.c
@@ -10,6 +10,7 @@
 //config:	bool "udhcpc6 (21 kb)"
 //config:	default y
 //config:	depends on FEATURE_IPV6
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	udhcpc6 is a DHCPv6 client
 //config:
--- 1/networking/zcip.c
+++ 2/networking/zcip.c
@@ -18,6 +18,7 @@
 //config:	default y
 //config:	select PLATFORM_LINUX
 //config:	select FEATURE_SYSLOG
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	ZCIP provides ZeroConf IPv4 address selection, according to RFC 3927.
 //config:	It's a daemon that allocates and defends a dynamically assigned
--- 1/procps/free.c
+++ 2/procps/free.c
@@ -10,6 +10,7 @@
 //config:	bool "free (3.1 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX #sysinfo()
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	free displays the total amount of free and used physical and swap
 //config:	memory in the system, as well as the buffers used by the kernel.
--- 1/procps/uptime.c
+++ 2/procps/uptime.c
@@ -15,6 +15,7 @@
 //config:	bool "uptime (3.7 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX #sysinfo()
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	uptime gives a one line display of the current time, how long
 //config:	the system has been running, how many users are currently logged
--- 1/sysklogd/klogd.c
+++ 2/sysklogd/klogd.c
@@ -34,6 +34,7 @@
 //config:	default y
 //config:	depends on KLOGD
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The klogd applet supports two interfaces for reading
 //config:	kernel messages. Linux provides the klogctl() interface
--- 1/util-linux/acpid.c
+++ 2/util-linux/acpid.c
@@ -10,6 +10,7 @@
 //config:	bool "acpid (9 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	acpid listens to ACPI events coming either in textual form from
 //config:	/proc/acpi/event (though it is marked deprecated it is still widely
--- 1/util-linux/blkdiscard.c
+++ 2/util-linux/blkdiscard.c
@@ -9,6 +9,7 @@
 //config:	bool "blkdiscard (4.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	blkdiscard discards sectors on a given device.
 
--- 1/util-linux/blkid.c
+++ 2/util-linux/blkid.c
@@ -10,6 +10,7 @@
 //config:	bool "blkid (12 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	select VOLUMEID
 //config:	help
 //config:	Lists labels and UUIDs of all filesystems.
--- 1/util-linux/blockdev.c
+++ 2/util-linux/blockdev.c
@@ -7,6 +7,7 @@
  */
 //config:config BLOCKDEV
 //config:	bool "blockdev (2.3 kb)"
+//config:	depends on BROKEN_ON_HURD
 //config:	default y
 //config:	help
 //config:	Performs some ioctls with block devices.
--- 1/util-linux/chrt.c
+++ 2/util-linux/chrt.c
@@ -8,6 +8,7 @@
 //config:config CHRT
 //config:	bool "chrt (4.7 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Manipulate real-time attributes of a process.
 //config:	This requires sched_{g,s}etparam support in your libc.
--- 1/util-linux/dmesg.c
+++ 2/util-linux/dmesg.c
@@ -12,6 +12,7 @@
 //config:	bool "dmesg (3.7 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	dmesg is used to examine or control the kernel ring buffer. When the
 //config:	Linux kernel prints messages to the system log, they are stored in
--- 1/util-linux/eject.c
+++ 2/util-linux/eject.c
@@ -16,6 +16,7 @@
 //config:	bool "eject (4 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Used to eject cdroms. (defaults to /dev/cdrom)
 //config:
--- 1/util-linux/fbset.c
+++ 2/util-linux/fbset.c
@@ -15,6 +15,7 @@
 //config:	bool "fbset (5.9 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	fbset is used to show or change the settings of a Linux frame buffer
 //config:	device. The frame buffer device provides a simple and unique
--- 1/util-linux/fdformat.c
+++ 2/util-linux/fdformat.c
@@ -9,6 +9,7 @@
 //config:	bool "fdformat (4.4 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	fdformat is used to low-level format a floppy disk.
 
--- 1/util-linux/fdisk.c
+++ 2/util-linux/fdisk.c
@@ -11,6 +11,7 @@
 //config:	bool "fdisk (37 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The fdisk utility is used to divide hard disks into one or more
 //config:	logical disks, which are generally called partitions. This utility
--- 1/util-linux/findfs.c
+++ 2/util-linux/findfs.c
@@ -11,6 +11,7 @@
 //config:	bool "findfs (12 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	select VOLUMEID
 //config:	help
 //config:	Prints the name of a filesystem with given label or UUID.
--- 1/util-linux/freeramdisk.c
+++ 2/util-linux/freeramdisk.c
@@ -12,6 +12,7 @@
 //config:	bool "fdflush (1.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	fdflush is only needed when changing media on slightly-broken
 //config:	removable media drives. It is used to make Linux believe that a
@@ -25,6 +26,7 @@
 //config:	bool "freeramdisk (1.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Linux allows you to create ramdisks. This utility allows you to
 //config:	delete them and completely free all memory that was used for the
--- 1/util-linux/fsfreeze.c
+++ 2/util-linux/fsfreeze.c
@@ -8,6 +8,7 @@
 //config:	bool "fsfreeze (3.5 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	select LONG_OPTS
 //config:	help
 //config:	Halt new accesses and flush writes on a mounted filesystem.
--- 1/util-linux/fstrim.c
+++ 2/util-linux/fstrim.c
@@ -11,6 +11,7 @@
 //config:	bool "fstrim (4.4 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Discard unused blocks on a mounted filesystem.
 
--- 1/util-linux/hwclock.c
+++ 2/util-linux/hwclock.c
@@ -10,6 +10,7 @@
 //config:	bool "hwclock (5.8 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The hwclock utility is used to read and set the hardware clock
 //config:	on a system. This is primarily used to set the current time on
--- 1/util-linux/ionice.c
+++ 2/util-linux/ionice.c
@@ -10,6 +10,7 @@
 //config:	bool "ionice (3.8 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Set/set program io scheduling class and priority
 //config:	Requires kernel >= 2.6.13
--- 1/util-linux/ipcrm.c
+++ 2/util-linux/ipcrm.c
@@ -10,6 +10,7 @@
 //config:config IPCRM
 //config:	bool "ipcrm (3.2 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The ipcrm utility allows the removal of System V interprocess
 //config:	communication (IPC) objects and the associated data structures
--- 1/util-linux/losetup.c
+++ 2/util-linux/losetup.c
@@ -10,6 +10,7 @@
 //config:	bool "losetup (5.5 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	losetup is used to associate or detach a loop device with a regular
 //config:	file or block device, and to query the status of a loop device. This
--- 1/util-linux/mdev.c
+++ 2/util-linux/mdev.c
@@ -11,6 +11,7 @@
 //config:	bool "mdev (17 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	mdev is a mini-udev implementation for dynamically creating device
 //config:	nodes in the /dev directory.
--- 1/util-linux/mkfs_ext2.c
+++ 2/util-linux/mkfs_ext2.c
@@ -11,6 +11,7 @@
 //config:	bool "mke2fs (10 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Utility to create EXT2 filesystems.
 //config:
@@ -18,6 +19,7 @@
 //config:	bool "mkfs.ext2 (10 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Alias to "mke2fs".
 
--- 1/util-linux/mkfs_reiser.c
+++ 2/util-linux/mkfs_reiser.c
@@ -10,6 +10,7 @@
 //config:	bool "mkfs_reiser"
 //config:	default n
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Utility to create ReiserFS filesystems.
 //config:	Note: this applet needs a lot of testing and polishing.
--- 1/util-linux/mkfs_vfat.c
+++ 2/util-linux/mkfs_vfat.c
@@ -11,6 +11,7 @@
 //config:	bool "mkdosfs (7.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Utility to create FAT32 filesystems.
 //config:
@@ -18,6 +19,7 @@
 //config:	bool "mkfs.vfat (7.2 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Alias to "mkdosfs".
 
--- 1/util-linux/mount.c
+++ 2/util-linux/mount.c
@@ -21,6 +21,7 @@
 //config:	bool "mount (23 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	All files and filesystems in Unix are arranged into one big directory
 //config:	tree. The 'mount' utility is used to graft a filesystem onto a
--- 1/util-linux/nsenter.c
+++ 2/util-linux/nsenter.c
@@ -10,6 +10,7 @@
 //config:	bool "nsenter (6.5 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Run program with namespaces of other processes.
 
--- 1/util-linux/pivot_root.c
+++ 2/util-linux/pivot_root.c
@@ -12,6 +12,7 @@
 //config:	bool "pivot_root (1.1 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The pivot_root utility swaps the mount points for the root filesystem
 //config:	with some other mounted filesystem. This allows you to do all sorts
--- 1/util-linux/rtcwake.c
+++ 2/util-linux/rtcwake.c
@@ -26,6 +26,7 @@
 //config:	bool "rtcwake (6.8 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Enter a system sleep state until specified wakeup time.
 
--- 1/util-linux/setarch.c
+++ 2/util-linux/setarch.c
@@ -10,6 +10,7 @@
 //config:	bool "setarch (3.6 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The linux32 utility is used to create a 32bit environment for the
 //config:	specified program (usually a shell). It only makes sense to have
@@ -20,6 +21,7 @@
 //config:	bool "linux32 (3.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Alias to "setarch linux32".
 //config:
@@ -27,6 +29,7 @@
 //config:	bool "linux64 (3.3 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Alias to "setarch linux64".
 
--- 1/util-linux/setpriv.c
+++ 2/util-linux/setpriv.c
@@ -11,6 +11,7 @@
 //config:	default y
 //config:	select PLATFORM_LINUX
 //config:	select LONG_OPTS
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Run a program with different Linux privilege settings.
 //config:	Requires kernel >= 3.5
@@ -27,6 +28,7 @@
 //config:	bool "Support capabilities"
 //config:	default y
 //config:	depends on SETPRIV
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Capabilities can be used to grant processes additional rights
 //config:	without the necessity to always execute as the root user.
--- 1/util-linux/swaponoff.c
+++ 2/util-linux/swaponoff.c
@@ -10,6 +10,7 @@
 //config:	bool "swapon (15 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Once you have created some swap space using 'mkswap', you also need
 //config:	to enable your swap space with the 'swapon' utility. The 'swapoff'
@@ -37,6 +38,7 @@
 //config:	bool "swapoff (14 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:
 //config:config FEATURE_SWAPONOFF_LABEL
 //config:	bool "Support specifying devices by label or UUID"
--- 1/util-linux/switch_root.c
+++ 2/util-linux/switch_root.c
@@ -10,6 +10,7 @@
 //config:	bool "switch_root (5.5 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	The switch_root utility is used from initramfs to select a new
 //config:	root device. Under initramfs, you have to use this instead of
--- 1/util-linux/taskset.c
+++ 2/util-linux/taskset.c
@@ -8,6 +8,7 @@
 //config:config TASKSET
 //config:	bool "taskset (4.2 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Retrieve or set a processes's CPU affinity.
 //config:	This requires sched_{g,s}etaffinity support in your libc.
--- 1/util-linux/uevent.c
+++ 2/util-linux/uevent.c
@@ -6,6 +6,7 @@
 //config:config UEVENT
 //config:	bool "uevent (3.1 kb)"
 //config:	default y
+//config:	depends on BROKEN_ON_HURD
 //config:	select PLATFORM_LINUX
 //config:	help
 //config:	uevent is a netlink listener for kernel uevent notifications
--- 1/util-linux/umount.c
+++ 2/util-linux/umount.c
@@ -11,6 +11,7 @@
 //config:	bool "umount (5.1 kb)"
 //config:	default y
 //config:	select PLATFORM_LINUX
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	When you want to remove a mounted filesystem from its current mount
 //config:	point, for example when you are shutting down the system, the
--- 1/util-linux/unshare.c
+++ 2/util-linux/unshare.c
@@ -12,6 +12,7 @@
 //config:	depends on !NOMMU
 //config:	select PLATFORM_LINUX
 //config:	select LONG_OPTS
+//config:	depends on BROKEN_ON_HURD
 //config:	help
 //config:	Run program with some namespaces unshared from parent.
 
