--- 1/applets/applet_tables.c
+++ 2/applets/applet_tables.c
@@ -20,6 +20,10 @@
 #undef ARRAY_SIZE
 #define ARRAY_SIZE(x) ((unsigned)(sizeof(x) / sizeof((x)[0])))
 
+#ifndef PATH_MAX
+#define PATH_MAX 1024
+#endif
+
 #include "../include/autoconf.h"
 #include "../include/applet_metadata.h"
 
--- 1/libbb/die_if_bad_username.c
+++ 2/libbb/die_if_bad_username.c
@@ -56,6 +56,8 @@
 	 * The maximum size of the login name is LOGIN_NAME_MAX
 	 * including the terminating null byte.
 	 */
+#ifdef LOGIN_NAME_MAX
 	if (name - start >= LOGIN_NAME_MAX)
 		bb_simple_error_msg_and_die("name is too long");
+#endif
 }
--- 1/mailutils/mail.c
+++ 2/mailutils/mail.c
@@ -6,7 +6,9 @@
  *
  * Licensed under GPLv2, see file LICENSE in this source tree.
  */
+#if ENABLE_SENDMAIL || ENABLE_POPMAILDIR
 #include <sys/prctl.h>
+#endif
 #include "libbb.h"
 #include "mail.h"
 
@@ -31,6 +33,7 @@
 #undef err
 }
 
+#if ENABLE_SENDMAIL || ENABLE_POPMAILDIR
 void FAST_FUNC launch_helper(const char **argv)
 {
 	// setup vanilla unidirectional pipes interchange
@@ -68,6 +71,7 @@
 
 	// parent goes on
 }
+#endif
 
 char* FAST_FUNC send_mail_command(const char *fmt, const char *param)
 {
--- 1/shell/shell_common.c
+++ 2/shell/shell_common.c
@@ -327,9 +327,11 @@
 static const struct limits limits_tbl[] = {
 	{ RLIMIT_CORE,		9,	}, // -c
 	{ RLIMIT_DATA,		10,	}, // -d
-	{ RLIMIT_NICE,		0,	}, // -e
 	{ RLIMIT_FSIZE,		9,	}, // -f
-#define LIMIT_F_IDX     3
+#define LIMIT_F_IDX     2
+#ifdef RLIMIT_NICE
+	{ RLIMIT_NICE,		0,	}, // -e
+#endif
 #ifdef RLIMIT_SIGPENDING
 	{ RLIMIT_SIGPENDING,	0,	}, // -i
 #endif
@@ -370,8 +372,10 @@
 static const char limits_help[] ALIGN1 =
 	"core file size (blocks)"          // -c
 	"\0""data seg size (kb)"           // -d
-	"\0""scheduling priority"          // -e
 	"\0""file size (blocks)"           // -f
+#ifdef RLIMIT_NICE
+	"\0""scheduling priority"          // -e
+#endif
 #ifdef RLIMIT_SIGPENDING
 	"\0""pending signals"              // -i
 #endif
@@ -410,8 +414,10 @@
 static const char limit_chars[] ALIGN1 =
 			"c"
 			"d"
-			"e"
 			"f"
+#ifdef RLIMIT_NICE
+			"e"
+#endif
 #ifdef RLIMIT_SIGPENDING
 			"i"
 #endif
@@ -451,8 +457,10 @@
 static const char ulimit_opt_string[] ALIGN1 = "-HSa"
 			"c::"
 			"d::"
-			"e::"
 			"f::"
+#ifdef RLIMIT_NICE
+			"e::"
+#endif
 #ifdef RLIMIT_SIGPENDING
 			"i::"
 #endif
